rfcs/2021-01-08-5843-encdoing-decoding-format-vector.md

```diff
@@ -0,0 +1,95 @@
+# RFC 5843 - 2021-01-08 - Encoding/Decoding for Vector to Vector Communication
+
+There has been an ongoing discussion of what format Vector-to-Vector communications should take. For some examples of issues related to the subject check out #5809, #5124, or #5341. This RFC is intended to propose a format for standardization of our vector-to-vector communications.
+
+## Scope
+
+- This RFC should cover the encoding format we intend to leverage
+- It should also cover any alterations that might need to be made to the expected data format
+
+## Motivation
+
+The question posed by @binarylogic in #5843 is a good definition for the motivation of this change: "How do we transmit data from one Vector instance to another while being mindful of the public contract we create?" Clearly sending metrics from one Vector instance to another should preserve all metric data upon ingestion downstream but there are still concerns around exposing a proprietary format to users in the context of these operations. This discussion continues to come up and we really need a canonincal and definitive decision around what shape this takes so that when we expand our components our contract is explicitly codified.
+
+## Internal Proposal
+
+From a high level my proposal is pretty simple. I'll dive into the rationale and context lower in the doc but lets just tee that up, shall we?
+
+I propose that if we're not searching explicitly for better perf around these messages and we're not explicitly searching for alternate tooling because we've decided that we wholly can't deal with the tooling overhead of protobufs we instead continue to use them with prost. I would also posit that if the concerns around exposing our private data format are bothersome enough, we should consider following an open spec for the data as a primary example: [opentelemetry](https://opentelemetry.io/).
```

Comment:
I would tend to agree. With regards to opentelemetry, while it does provide an interesting target for compatibility with other systems, it does not present the same model of data as our internal data. As such using it would require either an expensive import/export procedure (relative to a protobuf codec) or reworking our internal data model to work directly with OT data.

Reply:
Then I think we can safely remove OT from consideration.